parquet-thrift/src/main/java/org/apache/parquet/thrift/struct/ThriftTypeID.java

```diff
@@ -51,10 +51,15 @@ public enum ThriftTypeID {
   LIST (TType.LIST, true, ListType.class),
   ENUM (TType.ENUM, TType.I32, EnumType.class);
 
-  private static ThriftTypeID[] types = new ThriftTypeID[17];
+  private static final ThriftTypeID[] types;
   static {
+    types = new ThriftTypeID[18];
     for (ThriftTypeID t : ThriftTypeID.values()) {
-      types[t.thriftType] = t;
+      if (t.thriftType == -1) {
```

Comment:
I first thought it was because some class was moved, but this broke the initialization of the class: https://github.com/apache/thrift/blob/master/lib/java/src/main/java/org/apache/thrift/protocol/TType.java#L39-L40

Reply:
Good one, I've added it!